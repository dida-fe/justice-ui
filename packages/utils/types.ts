import { VNode, CreateElement, RenderContext } from 'vue';
import { InjectOptions, PropsDefinition } from 'vue/types/options';

export type EventHanlder = (eventName?: Event) => void;

export type ObjectIndex = {
  [key: string]: any;
};

export type ScopedSlot<Props = any> = (props?: Props) => VNode[] | VNode | undefined;

export type DefaultSlots = {
  default?: ScopedSlot;
};

export type ScopedSlots = DefaultSlots & {
  [key: string]: ScopedSlot | undefined;
};

export type ModelOptions = {
  prop?: string;
  event?: string;
};

export type DefaultProps = Record<string, any>;

export type FunctionComponent<Props = DefaultProps, PropDefs = PropsDefinition<Props>> = {
  (h: CreateElement, props: Props, slots: ScopedSlots, context: RenderContext<Props>):
    | VNode
    | undefined;
  props?: PropDefs;
  model?: ModelOptions;
  inject?: InjectOptions;
};
