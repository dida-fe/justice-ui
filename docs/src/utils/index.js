function iframeReady(iframe, callback) {
  const doc = iframe.contentDocument || iframe.contentWindow.document;
  const interval = () => {
    if (iframe.contentWindow.changePath) {
      callback();
    } else {
      setTimeout(() => {
        interval();
      }, 50);
    }
  };

  if (doc.readyState === 'complete') {
    interval();
  } else {
    iframe.onload = interval;
  }
}

const ua = navigator.userAgent.toLowerCase();
const isMobile = /ios|iphone|ipod|ipad|android/.test(ua);

function importAll(map, r) {
  r.keys().forEach(key => {
    map[key] = r(key);
  });
}

export {
  isMobile,
  importAll,
  iframeReady
};
