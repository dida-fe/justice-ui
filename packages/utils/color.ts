export const RED = '#f44';
export const BLUE = '#1989fa';
export const GREEN = '#07c160';
export const WHITE = '#fff';
export const GRAY = '#c9c9c9';
export const GRAY_DARK = '#969799';
