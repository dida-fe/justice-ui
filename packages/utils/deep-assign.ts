/* eslint-disable no-use-before-define */
import { isDef, isObj } from '.';
import { ObjectIndex } from './types';

const { hasOwnProperty } = Object.prototype;

function assignKey(to: ObjectIndex, from: ObjectIndex, key: string) {
  const val = from[key];

  if (!isDef(val)) {
    return;
  }

  if (!hasOwnProperty.call(to, key) || !isObj(val) || typeof val === 'function') {
    to[key] = val;
  } else {
    to[key] = deepAssign(Object(to[key]), from[key]);
  }
}

export function deepAssign(to: ObjectIndex, from: ObjectIndex): ObjectIndex {
  Object.keys(from).forEach(key => {
    assignKey(to, from, key);
  });

  return to;
}
