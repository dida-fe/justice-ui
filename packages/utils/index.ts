import Vue from 'vue';

export { use } from './use';
export { suffixPx } from './format/unit';

export const isServer: boolean = Vue.prototype.$isServer;

export function noop() {}

export function isDef(value: any): boolean {
  return value !== undefined && value !== null;
}

export function isObj(x: any): boolean {
  const type = typeof x;
  return x !== null && (type === 'object' || type === 'function');
}

export function get(object: any, path: string): any {
  const keys = path.split('.');
  let result = object;

  keys.forEach(key => {
    result = isDef(result[key]) ? result[key] : '';
  });

  return result;
}
