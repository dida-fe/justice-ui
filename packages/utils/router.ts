/**
 * Vue Router support
 */

import { RenderContext } from 'vue/types';
import VueRouter, { RawLocation } from 'vue-router/types';

export type RouteConfig = {
  url?: string;
  to?: RawLocation;
  replace?: boolean;
};

export function route(router: VueRouter, config: RouteConfig) {
  const { to, url, replace } = config;
  if (to && router) {
    router[replace ? 'replace' : 'push'](to);
  } else if (url) {
    replace ? location.replace(url) : (location.href = url);
  }
}

export function functionalRoute(context: RenderContext) {
  route(context.parent && context.parent.$router, context.props);
}

export type RouteProps = {
  url?: string,
  replace?: boolean;
  to?: RawLocation;
}

export const routeProps = {
  url: String,
  replace: Boolean,
  to: [String, Object]
};
