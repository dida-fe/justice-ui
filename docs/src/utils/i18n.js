// component mixin
import { get } from '../../../packages/utils';
import { camelize } from '../../../packages/utils/format/string';

export default {
  computed: {
    $t() {
      const { name } = this.$options;
      const prefix = name ? camelize(name) + '.' : '';
      const messages = this.$vantMessages[this.$vantLang];

      return (path, ...args) => {
        const message = get(messages, prefix + path) || get(messages, path);
        return typeof message === 'function' ? message(...args) : message;
      };
    }
  }
};
