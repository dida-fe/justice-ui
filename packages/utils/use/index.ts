import { useBEM } from './bem';
import { useSFC } from './sfc';
import { useI18N } from './i18n';

type UseReturn = [
  ReturnType<typeof useSFC>,
  ReturnType<typeof useBEM>,
  ReturnType<typeof useI18N>
];

export function use(name: string): UseReturn {
  name = 'justice-' + name;
  return [useSFC(name), useBEM(name), useI18N(name)];
}
