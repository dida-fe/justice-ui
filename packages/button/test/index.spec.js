import { mount } from '../../../test/utils';
import Button from '..';

test('loading size', () => {
  const wrapper = mount(Button, {
    propsData: {
      loading: true,
      loadingSize: '10px'
    }
  });
  expect(wrapper).toMatchSnapshot();
});

test('click event', () => {
  const onClick = jest.fn();
  const wrapper = mount(Button, {
    context: {
      on: {
        click: onClick
      }
    }
  });

  wrapper.trigger('click');
  expect(onClick).toHaveBeenCalled();
});

test('not trigger click event when disabled', () => {
  const onClick = jest.fn();
  const wrapper = mount(Button, {
    propsData: {
      disabled: true
    },
    context: {
      on: {
        click: onClick
      }
    }
  });

  wrapper.trigger('click');
  expect(onClick).toHaveBeenCalledTimes(0);
});

test('not trigger click event when loading', () => {
  const onClick = jest.fn();
  const wrapper = mount(Button, {
    propsData: {
      loading: true
    },
    context: {
      on: {
        click: onClick
      }
    }
  });

  wrapper.trigger('click');
  expect(onClick).toHaveBeenCalledTimes(0);
});

test('touchstart event', () => {
  const onTouchstart = jest.fn();
  const wrapper = mount(Button, {
    context: {
      on: {
        touchstart: onTouchstart
      }
    }
  });

  wrapper.trigger('touchstart');
  expect(onTouchstart).toHaveBeenCalled();
});
