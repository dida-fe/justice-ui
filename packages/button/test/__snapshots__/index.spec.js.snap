// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading size 1`] = `
<button class="justice-button justice-button--default justice-button--normal">
  <div class="van-loading van-loading--circular justice-button__loading"><span class="van-loading__spinner van-loading__spinner--circular" style="color: rgb(201, 201, 201); width: 10px; height: 10px;"><svg viewBox="25 25 50 50" class="van-loading__circular"><circle cx="50" cy="50" r="20" fill="none"></circle></svg></span></div>
</button>
`;
